cmd/state-mcp/internal/toolregistry/tools.go

```diff
@@ -37,4 +41,86 @@ func HelloWorldTool() Tool {
 			), nil
 		},
 	}
-}
\ No newline at end of file
+}
+
+type BuildNodesResponse struct {
+	Commit struct {
+		Build struct {
+			Nodes []BuildNode `json:"nodes"`
+		} `json:"build"`
+	} `json:"commit"`
+}
+
+type BuildNode struct {
+	Typename            string `json:"__typename"`
+	Name                string `json:"name"`
+	Namespace           string `json:"namespace"`
+	Version             string `json:"version"`
+	DisplayName         string `json:"displayName"`
+	LogURL              string `json:"logURL"`
+	Status              string `json:"status"`
+	LastBuildFinishedAt string `json:"lastBuildFinishedAt"`
+}
+
+func ProjectErrorsTool() Tool {
+	return Tool{
+		Category: ToolCategoryDebug,
+		Tool: mcp.NewTool(
+			"projecterrors",
+			mcp.WithDescription("Project errors tool - shows errors for a project"),
+			mcp.WithString("namespace", mcp.Description("Project namespace in format 'owner/project'")),
+		),
+		Handler: func(ctx context.Context, p *primer.Values, mcpRequest mcp.CallToolRequest) (*mcp.CallToolResult, error) {
+			namespace, _ := mcpRequest.RequireString("namespace")
+
+			ns, err := project.ParseNamespace(namespace)
+			if err != nil {
+				return mcp.NewToolResultError("Invalid namespace format. Use 'owner/project' format."), nil
+			}
+
+			runner := graphql.New(p, api.ServiceBuildPlanner)
+			request := &graphql.Request{
+				QueryStr: `query($organization: String!, $project: String!) {
+								project(organization: $organization, project: $project) {
+									... on Project {
+										commit {
+											... on Commit {
+												build {
+													... on Build {
+														nodes {
+															... on Source {
+																__typename, name, version, namespace
+															}
+															... on ArtifactPermanentlyFailed {
+																__typename, displayName, logURL, status, lastBuildFinishedAt
+															}
+														}
+													}
+												}
+											}
+										}
+									}
+								}
+							}`,
+				QueryVars: map[string]interface{}{
+					"organization": ns.Owner,
+					"project":      ns.Project,
+				},
+			}
+			response := BuildNodesResponse{}
+			err = runner.Run(request, &response)
+			if err != nil {
+				return mcp.NewToolResultError(err.Error()), nil
+			}
+
+			// at this point we can handle request to create the final output
+			// OR just return the full response as text
+			jsonBytes, _ := json.Marshal(response)
```

Comment:
We should never drop errors, unless the code in question is for some quick one off use-case. Error handling in Go can be annoying, but for better or worse it is a core part of the language and dropping them often leads to needle in the haystack type problems.